File: BigData3/final-project/src/test/java/univ/bigdata/course/MovieTest.java
package univ.bigdata.course;

import org.junit.Assert;
import org.junit.Test;
import univ.bigdata.course.internal.MoviesFunctions;
import univ.bigdata.course.part1.movie.InternalMovie;

import java.util.ArrayList;
import java.util.List;

import static univ.bigdata.course.TestBuilders.*;

public class MovieTest {

    @Test
    public void testMovieAvg() {
        InternalMovie movie1 = scoredMovie("movie1", 1, 5, 3, 4, 1, 1);
        double expectedAvg = (double) (1 + 5 + 3 + 4 + 1 + 1) / 6;
        Assert.assertEquals(expectedAvg, movie1.avgScore(), 1e-15);
    }

    @Test
    public void testMovieAvgOneReview() {
        InternalMovie movieNoReviews = scoredMovie("movie2", 3);
        double expectedAvg = 3;
        Assert.assertEquals(expectedAvg, movieNoReviews.avgScore(), 1e-15);
    }

    @Test
    public void testMostPopularMovieReviewedByKUsers () {
        InternalMovie movie1 = scoredMovie("Yuval", 1, 5, 3, 3, 5, 9, 10);
        InternalMovie movie2 = scoredMovie("Tony", 0, 0, 4, 3, 1);
        List<InternalMovie> vika = new ArrayList<>(2);
        vika.add(movie1);
        vika.add(movie2);
        MoviesFunctions storage = new MoviesFunctions(vika);
        String id = storage.mostPopularMovieReviewedByKUsers(6);
        Assert.assertEquals(id, movie1.movieId);
    }

    @Test
    public void testMoviePrecentile()
    {
        List<InternalMovie> movies = new ArrayList<>();
        InternalMovie movie1 = scoredMovie("AA", 5, 7, 9, 8, 9, 9, 9);
        InternalMovie movie2 = scoredMovie("BB", 8);
        InternalMovie movie3 = scoredMovie("cc", 10);
        InternalMovie movie4 = scoredMovie("dd", 10);
        movies.add(movie1);
        movies.add(movie2);
        movies.add(movie3);
        movies.add(movie4);
        MoviesFunctions storage = new MoviesFunctions(movies);
        List<InternalMovie> t = storage.getMoviesPercentile(75);
        Assert.assertEquals(movie3.movieId, t.get(0).movieId);
    }

}

File: BigData3/final-project/src/test/scala/univ/bigdata/course/MovieTest.scala
package univ.bigdata.course

import org.junit.{Assert, Test}
import TestBuilders._
import univ.bigdata.course.part1.MoviesFunctions

class MovieTest {
  @Test
  def testMovieAvg() {
    val movie1 = scoredMovie("movie1", 1, 5, 3, 4, 1, 1)
    val expectedAvg: Double = (1 + 5 + 3 + 4 + 1 + 1).toDouble / 6
    Assert.assertEquals(expectedAvg, movie1.avgScore, Epsilon)
  }

  @Test
  def testMovieAvgOneReview() {
    val movieNoReviews = scoredMovie("movie2", 3)
    val expectedAvg: Double = 3
    Assert.assertEquals(expectedAvg, movieNoReviews.avgScore, Epsilon)
  }

  @Test
  def testMostPopularMovieReviewedByKUsers() {
    val movie1  = scoredMovie("Yuval", 1, 5, 3, 3, 5, 9, 10)
    val movie2 = scoredMovie("Tony", 0, 0, 4, 3, 1)
    val movies = moviesRDD(movie1, movie2)
    val mostReviewedMovie = MoviesFunctions.mostPopularMovieReviewedByKUsers(movies, 6)
    Assert.assertEquals(movie1, mostReviewedMovie)
  }

  @Test
  def testMoviePercentile() {
    val movie1 = scoredMovie("AA", 5, 7, 9, 8, 9, 9, 9)
    val movie2 = scoredMovie("BB", 8)
    val movie3 = scoredMovie("cc", 10)
    val movie4 = scoredMovie("dd", 10)
    val movies = moviesRDD(movie1, movie2, movie3, movie4)
    val percentile = MoviesFunctions.getMoviesPercentile(movies, 75)
    Assert.assertEquals(Vector(movie3), percentile)
  }

}
